bash/sanity_checks.bash

```diff
@@ -299,12 +299,16 @@ function __static__Set_Software_Input_Data_File_If_Not_Set_By_User()
             printf -v filename '%s/%s' \
                 "${HYBRID_software_output_directory[${relative_key}]}" \
                 "${HYBRID_software_default_input_filename[${key}]}"
-        else
+        elif [[ "${filename}" =~ / ]]; then
             if Element_In_Array_Equals_To "${relative_key}" "${HYBRID_given_software_sections[@]}"; then
                 exit_code=${HYBRID_fatal_logic_error} Print_Fatal_And_Exit \
                     'Requesting custom ' --emph "${key}" ' input file although executing ' \
                     --emph "${relative_key}" ' with default output name.'
             fi
+        else
+            printf -v filename '%s/%s' \
+                "${HYBRID_software_output_directory[${relative_key}]}" \
+                "${filename}"
```

Comment:
Since I am not totally sure that this input is validated at any point, we could probably catch here basic misuse. In general, the Hybrid handler at some points calls
```bash
Ensure_Given_Files_Exist "${HYBRID_software_input_file[...]}"
```
and this will fail if e.g. the provided input file is like a directory. At the moment the user can provide `.` or `..` as input file. This will fall into this `else` and I am wondering whether we should avoid that... 🤔